libmuscle/python/libmuscle/instance.py

```diff
@@ -126,7 +183,7 @@ def __init__(
         self._set_local_log_level()
         self._set_remote_log_level()
 
-    def reuse_instance(self, apply_overlay: bool = True) -> bool:
+    def reuse_instance(self, apply_overlay: Optional[bool] = None) -> bool:
```

Comment:
Given the API change, should we check this for compatibility with the flags? Or warn if it's specified that it will be ignored? I doubt anyone is using it, so I think it's okay if it doesn't work. Actually, then we could just remove it.